### PR TITLE
Assume that snapshots are local:

### PR DESCRIPTION
**Pull request by stickycode**
Because I never publish snapshots, there is no metadata to resolve them.
Occasionally you need to build locally and resolve snaphots with no metadata, this change just assumes they exist locally.

### CODE REVIEW

File: src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java [hunk 1]
```diff
@@ -115,8 +148,20 @@ Artifact lookupArtifact(DefaultArtifact artifact) {
...
       artifact.getExtension(),
       version.toString());
-    ArtifactRequest request = new ArtifactRequest(fetch, repositories, null);
-    return resolve(request).getArtifact();
+    return new ArtifactRequest(fetch, repositories, null);
+  }
+
+  Properties getProjectProperties() {
+    return project.getProperties();
+  }
```
Comment (fnzprac) [inline src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java:156]: Why?

Comment (fredcooke) [inline src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java:156]: Why?

File: src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java [hunk 2]
```diff
@@ -181,20 +226,19 @@ private void unpack(Artifact artifact) {
...
-      return repository.resolveArtifact(session, request);
-    }
-    catch (ArtifactResolutionException e) {
-      throw new RuntimeException(e);
+  private Version highestVersion(Artifact artifact) {
+    if (assumeSnapshotsAreLocal()) {
+      if (artifact.getVersion().endsWith("-SNAPSHOT"))
+        return new VersionImplementation().withVersion(artifact);
+
+      if (artifact.getVersion().endsWith("-SNAPSHOT]") && artifact.getVersion().startsWith("["))
```
Comment (fredcooke) [inline src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java:234]: If I'm reading this right you're assuming a ] doesn't come after the [ - that might not be healthy to do?

File: src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java (hunk 1 above)
Comment (stickycode) [reply src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java:156]: exposed for testing, should probably put the annotations or use Mockito

File: src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java (hunk 2 above)
Comment (stickycode) [reply src/main/java/net/stickycode/plugin/shifty/ShiftyFetchMojo.java:234]: its just checking to see if the version is [...-SNAPSHOT] i.e. a fixed snapshot version

**Pull request merged**